### DISCUSSION TITLE
Send attachments to external db

### DISCUSSION THREAD

**Discussion by adamjehu** [Budibase/budibase#2972] [Help] [answered]
Hi! 
For my app I am connecting to a MySQL database. I have no issues querying and writing to it from the app except for pushing attachments to a BLOB field in the database. I inserted an Attachment element and added the binding to the BLOB column within the Save button's actions similar to:
Column: MyBlobColumn
Value: {{ Form.Fields.attachElement }}
I get no errors when hitting save on the form, but when I go to the MySQL table, the BLOB field is empty - all others are populated. Also I should note, looking at the MySQL table under Sources, the BLOB column contains {"type":"Buffer","data":[]} for all rows.

**Comment by mjashanks** [comment, marked as answer]
Hey @adamjehu - unfortunately, this is just not possible at the moment. Attachments only work with internal tables, and there is no way to link them.

**Reply by adamjehu** [reply]
Hi @mjashanks thank you for your quick reply. Is there another way you can suggest capturing the image using the form but then perhaps having it saved to a folder or network location?

**Reply by mjashanks** [reply]
Can't think of any other way, sorry. 

I'd really like to get attachments working with SQL, but can't give you a date for that.

**Reply by adamjehu** [reply]
@mjashanks thanks for the info and trying to figure out a solution for me. I hope to see this as a feature or as part of the automation tasks in the future.

**Reply by dncpax** [reply]
can you put an attachment in an internal temp table and use automation to send it to an external db or even to a web service that writes the attachment to the external db?

**Reply by mjashanks** [reply]
There is no way to Send attachments outside of the platform at the moment.

**Reply by dncpax** [reply]
@mjashanks can you chain an internal sql query with a rest request? put the result of the sql query in the rest request? could we encode64 a binary from the sql and send it to a rest backend? i have not worked with automations yet so just curious...

**Reply by mjashanks** [reply]
yes - that should be possible.

**Comment by stephlegrand** [comment]
I am trying to achieve the same thing - I need to automate a workflow and it means that some of the attachments need to be exported to a CRM system.

I can get to the link of the first attachments and so I can automate download from the other side via a HTTP query.

The only problem is that only the link to the first attachment is available.

**Reply by mjashanks** [reply]
I'm not sure that I fully understand you, but I think it should be covered by #3173.

This will let you display links for all attachements.

**Reply by stephlegrand** [reply]
Sorry if I was not clear. This sounds perfect anyway.

Thanks a lot!
